### ISSUE TITLE
Wrong / Corrupted Data loaded 

### ISSUE THREAD

**Issue by aruffj** [diasurgical/devilutionX#1923]
**Base Information**:
Platform: Win64 / Linux
Version: 1.2.1 (Release Version)

**Issue**:
Trying to load Level 10 results in an not correctly loaded area. 

**Steps to reproduce**:
 - Load savegame
 - Move to Level 10

**Note**:
 - This is an "converted" savegame (.sv => .hsv)
 - I haven't explored level 10 before

**Video and files**:
I took a short (~20 Seconds) recording which can be found unlisted on youtube [here](https://youtu.be/PVnCjsxej_0). Attached are also the savegame and diablo.ini. Both can befound [here](https://github.com/diasurgical/devilutionX/files/6450390/devilutionx.zip).

**Comment by obligaron** [comment]
I checked the save game. Level 1-16 are set as visited.
I also checked if the visited flag stays when switching from diablo to hellfire. This is the case.
If the dungeon layout can differ between diablo and hellfire then this could explain the shown result. But I'm not familiar enough with the dungeon generator to give a correct answer on this. Does someone know?
Alternative this could be a reroll bug.

**Comment by AJenbo** [comment]
Level 1-4 will generate different layout between Hellfire and Diablo, 5-16 generates the same layout.

Did you check the same levels in Hellfire 1.01 and Diablo 1.09?

Anvil of Fury is found on level 10 so a reroll could potentially be a posibility. Anvil of Fury is in the same group as Zhar and the Mushroom quest.

**Comment by obligaron** [comment]
> Did you check the same levels in Hellfire 1.01 and Diablo 1.09?

The result in vanilla and hellfire is the same as devilutionX.
(For testing vanilla, I converted the save game back to diablo/.sv)

> Anvil of Fury is found on level 10 so a reroll could potentially be a posibility. Anvil of Fury is in the same group as Zhar and the Mushroom quest.

In devilutionX I activated the quest. The level layout changes but is still broken / differs from save objects (as the original issue).
Also after activating the quest I got some errors rendering dead monsters.

I'm still unsure how we got to this point. :thinking:
